specification/network/resource-manager/Microsoft.Network/stable/2021-05-01/firewallPolicy.json

```diff
@@ -1464,6 +1714,236 @@
           "description": "The workspace Id for Firewall Policy Insights."
         }
       }
+    },
+    "IDPSQueryObject": {
+      "description": "Will describe the query to run against the IDPS signatures DB",
+      "type": "object",
+      "properties": {
+        "filters": {
+          "type": "object",
+          "description": "Contain all filters names and values",
+          "$ref": "#/definitions/Filters"
+        },
+        "search": {
+          "type": "string",
+          "description": "Search term in all columns"
+        },
+        "orderBy": {
+          "type": "object",
+          "$ref": "#/definitions/OrderBy",
+          "description": "Column to sort response by"
+        },
+        "resultsPerPage": {
+          "type": "integer",
+          "format": "int32",
+          "minimum": 1,
+          "maximum": 1000,
+          "description": "The number of the results to return in each page"
+        },
+        "skip": {
+          "type": "integer",
+          "format": "int32",
+          "description": "The number of records matching the filter to skip"
+        }
+      }
+    },
+    "Filters": {
+      "description": "Describers the filters to filter response by",
+      "type": "array",
+      "items": {
+        "$ref": "#/definitions/FilterItems"
+      }
+    },
+    "FilterItems": {
+      "description": "Will contain the filter name and values to operate on",
+      "type": "object",
+      "properties": {
+        "field": {
+          "type": "string",
+          "description": "The name of the field we would like to filter"
+        },
+        "values": {
+          "type": "array",
+          "description": "List of values to filter the current field by",
+          "items": {
+            "type": "string",
+            "description": "Value of the field to filter by"
+          }
+        }
+      }
+    },
+    "OrderBy": {
+      "description": "Describes a column to sort",
+      "type": "object",
+      "properties": {
+        "field": {
+          "type": "string",
+          "description": "Describes the actual column name to sort by"
+        },
+        "asc": {
```

Comment:
>asc [](http://example.com/codeflow?start=9&length=3)

ARM generally recommends against boolean in favor of string enum. In this case it would be something like "order: [Descending, Ascending].

Reply:
done